sdk/search/search-documents/CHANGELOG.md

```diff
@@ -9,13 +9,41 @@
 ### Breaking Changes
 
 - Update `SearchIndexerDataNoneIdentity.odatatype`. [#23495](https://github.com/Azure/azure-sdk-for-js/pull/23495)
+- Fix `SearchClient.getDocument` returning documents that contain fields which were not selected [#23627](https://github.com/Azure/azure-sdk-for-js/pull/23627)
+- Add type narrowing to returned documents [#23627](https://github.com/Azure/azure-sdk-for-js/pull/23627)
+  - Add breaking type parameters to these types:
```

Comment:
Just out of curiosity, how big do we expect this breaking change to be? Is it something that everyone who updates will run into breaks? Or is it more like a couple of corner scenarios where users would get broken?

Also if I'm in a broken situation, how easy/hard is to fix my way out of the broken state?

Reply:
The returned documents should be assignable to the wide document type, but only if the key value is non-nullable as specified in the JSDoc for `SearchClient`'s constructor. If users have it declared declared as `object`, it's not breaking. For the affected type parameters, it's just a matter of adding the `Model` type parameter for named instances of those types. I think both of these cases are rare. All in all, I wouldn't expect it to break for most people, and it shouldn't be difficult to fix.